src/main/scala/scala/collection/decorators/IteratorDecorator.scala

```diff
@@ -5,6 +5,20 @@ import scala.annotation.tailrec
 
 class IteratorDecorator[A](val `this`: Iterator[A]) extends AnyVal {
 
+  /**
+    * Inserts a separator value between each element.
+    *
+    * {{{
+    *   Iterator(1, 2, 3).intersperse(0) === Iterator(1, 0, 2, 0, 3)
+    *   Iterator('a', 'b', 'c').intersperse(',') === Iterator('a', ',', 'b', ',', 'c')
+    *   Iterator('a').intersperse(',') === Iterator('a')
+    *   Iterator().intersperse(',') === Iterator()
+    * }}}
+    *
+    * @param sep the separator value.
+    * @return    The resulting iterator contains all elements from the source iterator, separated by the `sep` value.
+    * @note      Reuse: $consumesIterator
```

Comment:
I think here we should use `$consumeAndProducesIterator` instead

Reply:
Yeah, already fixed that :)